src/analyses/mutexEventsAnalysis.ml

```diff
@@ -0,0 +1,153 @@
+(** Mutex analysis. *)
+
+module M = Messages
+module Addr = ValueDomain.Addr
+module Lockset = LockDomain.Lockset
+module Mutexes = LockDomain.Mutexes
+module LF = LibraryFunctions
+open Prelude.Ana
+open Analyses
+open GobConfig
+
+let big_kernel_lock = LockDomain.Addr.from_var (Goblintutil.create_var (makeGlobalVar "[big kernel lock]" intType))
+let console_sem = LockDomain.Addr.from_var (Goblintutil.create_var (makeGlobalVar "[console semaphore]" intType))
+let verifier_atomic = LockDomain.Addr.from_var (Goblintutil.create_var (makeGlobalVar "[__VERIFIER_atomic]" intType))
+
+module Spec: MCPSpec =
+struct
+  include UnitAnalysis.Spec
+  let name () = "mutexEvents"
+
+
+  (* NB! Currently we care only about concrete indexes. Base (seeing only a int domain
+     element) answers with the string "unknown" on all non-concrete cases. *)
+  let rec conv_offset x =
+    match x with
+    | `NoOffset    -> `NoOffset
+    | `Index (Const (CInt (i,_,s)),o) -> `Index (IntDomain.of_const (i,Cilfacade.ptrdiff_ikind (),s), conv_offset o)
+    | `Index (_,o) -> `Index (ValueDomain.IndexDomain.top (), conv_offset o)
+    | `Field (f,o) -> `Field (f, conv_offset o)
+
+  let rec conv_offset_inv = function
+    | `NoOffset -> `NoOffset
+    | `Field (f, o) -> `Field (f, conv_offset_inv o)
+    (* TODO: better indices handling *)
```

Comment:
Should we do this here then? It should be trivial to go back to `Const (CInt (i,_,s))` here, right?

Reply:
Done in MutexAnalysis, symmetrically to `conv_offset`.